spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Linear.scala

```diff
@@ -81,6 +80,12 @@ class Linear[T: ClassTag](
     zeroGradParameters()
   }
 
+  override def computeOutputShape(inputShape: Shape): Shape = {
+    val input = inputShape.toSingle().toArray
```

Comment:
override the update method?

Reply:
`updateoutput`? I guess it can be in refactor in a following PR as it would affect the core computation.